setup.py

```diff
@@ -76,6 +82,13 @@ def read_version_from_package_xml():
         'Programming Language :: Python :: 3.7',
         'Topic :: Software Development',
     ],
+    # NOTE: when we upgrade to pip 10, we will need to update to PEP 508 syntax
+    dependency_links=[
+        ament_lint_url('ament_lint'),
```

Comment:
It looks like the GH Action can't use this.
[StackOverflow](https://stackoverflow.com/a/54794506) suggests only using the `test_require`

Reply:
`tests_require` can't use URLs. it seems to be that setuptools can't parse the `subdirectory` portion of the URL, it's not fully pip-compliant or something. i'll try a tarball option